runtime/sys.c

```diff
@@ -574,7 +581,16 @@ int caml_unix_random_seed(intnat data[16])
   int fd;
   int n = 0;
 
-  /* Try /dev/urandom first */
+  /* Try kernel entropy first */
+#ifdef HAS_GETENTROPY
+  if (getentropy(data, 12) == 0) {
+    /* If the kernel succeeded, we now have 96 bits
+       of good random data and can stop here. */
+    return 12;
```

Comment:
Watch out: `data` is an array of integers... If you look at the /dev/urandom case below, the 12 random bytes are expanded to 12 integers.  It was necessary with the previous implementation of Random, based on a lagged Fibonacci generator, and is no longer strictly necessary, but it would be nicer to stay consistent with the /dev/urandom case.

Reply:
The comment on lines 587 is wrong: we have 96 **bytes** (768 bits) of randomness, not 96 **bits**. Assuming 64-bit `intnat`.

Indeed this looks overkill (the PRNG has 256 bits of internal state). Still, the /dev/urandom code below provides only 96 bits, and it would be better to provide 256.

Maybe the /dev/urandom code should also be fixed to provide 256 bits.

Reply:
No.  If you read 12 good random bytes, then expand each byte into a 64-bit integer, you still only have 12 * 8 = 96 bits of entropy.

Reply:
`getentropy` takes a `void *` and a count in bytes.  Here we're passing an array of 64-bit integers but it still fills only the first integer and half of the second integer.  This isn't great, and that's why it should better be expanded like the /dev/urandom read does.

Reply:
Thank you. I didn't notice the type differences there.